src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -576,15 +607,20 @@
     <!-- Extract repo symbols tarball -->
     <MakeDir Directories="$(IntermediateSymbolsRepoDir)" />
     <Exec Command="tar -xzf %(RepoSymbolsArchive.Identity) -C $(IntermediateSymbolsRepoDir)" />
+
+    <!-- Delete repo symbols archive in assets folder -->
+    <Delete Condition="'%(RepoSymbolsArchive.Identity)' == '$(RepoAssetsSymbolsArchive)'" Files="%(RepoSymbolsArchive.Identity)" />
```

Comment:
Let's not copy something to the assets folder to just then later delete it again. It might make more sense to not publish the repo symbols archive file. We could list it in the manifest file (i.e. as additional build output) and then read that here or tell the inner/outer repo publish infrastructure to copy it to a given location.

Reply:
I think it would be best to not include it in manifest at all, as the files will be copied to a different folder, not under `assets`.

How about another property that we pass to inner build that publish infra uses to copy the symbols archive?

@mmitche

Reply:
I thought about this some more. I'm going to stop deleting symbols archives and I'm going to update the metadata for these assets so they are published to a sub-directory of assets root, i.e. `Symbols`. We only publish these on Linux.

I want to avoid adding custom publishing infra for this.

Reply:
Anything under artifacts/assets is going to be uploaded to AzDO and maybe even the blob-feed. Is that what we want here? These manifests sound like intermediates to me and not final outputs.

Reply:
I'm not sure. We were trying to avoid doing any custom copying from inner-build.

Symbols packages would account for 2% of assets directory size in Linux builds. We do not produce them on other systems.

Reply:
I think I would treat this similar to the generated repo manifest which your change now gets copied to a repo specific intermediate location. I think the same should happen for the repo symbol package. Only final (regardless of shipping vs non-shipping) outputs should go into artifacts/packages and artifacts/assets.

Reply:
I see. So, adding a new property to tell the inner-build where to copy this specific archive would be the goal? I see the point in this and this can also be done in `AfterSourceBuild.proj`, to not add unnecessary non-publishing infra to arcade's Publish.proj. I'd like to also hear what @mmitche thinks about this.

Reply:
Right, but I don't have a strong opinion on this as it's source-build only. That said, it would be great to honor the distinction between final and intermediate outputs.

Aside from that, I wonder why we even create an archive file. If we just directly copy into a folder during the repo publishing step, then we save the time to compress, un-compress and then delete the archive file.

Reply:
We used to create archive file, for packaging in intermediate package. That is still done in repo's source-build. VMR build does not really need the archive since we got rid of intermediate packages in that scenario.

Reply:
> I see. So, adding a new property to tell the inner-build where to copy this specific archive would be the goal? I see the point in this and this can also be done in `AfterSourceBuild.proj`, to not add unnecessary non-publishing infra to arcade's Publish.proj. I'd like to also hear what @mmitche thinks about this.

I think putting this in AfterSourceBuild makes sense to me. You may need to put it under a source build only switch, as IIRC AfterSourceBUild.proj is still used in MSFT VMR builds.

Reply:
Nikola filed https://github.com/dotnet/source-build/issues/4225 to track this improvement